### PR TITLE
Bringup Pi_0 Pytorch model

### PR DESCRIPTION
**Pull request by ashokkumarkannan1**
### Ticket
Link to Github Issue - https://github.com/tenstorrent/tt-xla/issues/2979

### Problem description
to bringup Pi_0 model

### What's changed
added Pi_0 pytorch model

### Checklist
- [x] New/Existing tests provide coverage for changes

### Logs
- [xla_inference_test_base.log](https://github.com/user-attachments/files/25284899/xla_inference_test_base.log)
- [xla_inference_test_libero.log](https://github.com/user-attachments/files/25284900/xla_inference_test_libero.log)

### CODE REVIEW

**Review by meenakshiramanathan1 — APPROVED** [review]

File: pi_0/pytorch/loader.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        )
+
+    def load_model(self, dtype_override=None):
+        """Load and return the Pi-0 model instance with default settings.
+
+        Returns:
+            torch.nn.Module: The Pi-0 Policy instance.
+        """
+        from lerobot.policies.pi0 import PI0Policy
+        from .src import model
```
Comment (ppadjinTT) [inline pi_0/pytorch/loader.py]: due to the global PI0Policy object I mentioned earlier, it is not clear whether you are using the original PI0Policy object, or is it getting overwritten by the one from .src.model? Please change to from `.src.model import get_custom_pi0_policy` and then I assume this `from lerobot.policies.pi0 import PI0Policy` import will be unnessesary

File: pi_0/pytorch/src/model.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    lang_tokens = batch["observation.language.tokens"]
+    lang_masks = batch["observation.language.attention_mask"]
+
+    state = self.prepare_state(batch)
+
+    return images, img_masks, lang_tokens, lang_masks, state
+
+
+@torch.no_grad()
+def forward(
```
[the remark this answers is not in the document]
Comment (ashokkumarkannan1) [reply pi_0/pytorch/src/model.py]: - The [select_action](https://github.com/huggingface/lerobot/blob/main/src/lerobot/policies/pi0/modeling_pi0.py#L1229) method is responsible for inference (as described in [source](https://huggingface.co/lerobot/pi0_libero_base#load-model--dataset-run-select_action)). The [Policy.forward](https://github.com/huggingface/lerobot/blob/main/src/lerobot/policies/pi0/modeling_pi0.py#L1264) method in LeRobot is only used during training to compute the loss and is not part of the inference path.

- Since our use case targets inference, we needed to override forward to incorporate the `select_action` logic, ensuring the correct execution flow during model inference. (as already mentioned in the  forward method [docstring](https://github.com/tenstorrent/tt-forge-models/pull/476/changes/BASE..ac131bc107e90d8fe6781ff5f1cb68cd96aa2e32#diff-6e6bfcb85c8cba189efb0b8b07ac3a1e2eba541a3a9c59fe7470694758b37f52R53))
 
- Additionally, the steps inside `preprocess_for_sampling`, which are normally invoked within `select_action`, have been separated and applied directly during the `load_inputs` stage. This is because those steps are specifically responsible for preprocessing a batch before action sampling.

File: pi_0/pytorch/loader.py (hunk 1 above)
Comment (ashokkumarkannan1) [reply pi_0/pytorch/loader.py]: - I’ve updated the `load_model` function to instantiate the model using `get_custom_pi0_policy`, which internally handles importing the original `PI0Policy` and applying the required overrides.

- As a result, the direct import `from lerobot.policies.pi0 import PI0Policy` has been removed from this file to avoid any confusion about which policy object is being used.
